src/main/kotlin/com/openlattice/mechanic/upgrades/V3StudyMigrationUpgrade.kt

```diff
@@ -0,0 +1,178 @@
+
+    private val STRING_ID_FQN = FullQualifiedName( "general.stringid" )
+    private val FULL_NAME_FQN = FullQualifiedName( "general.fullname" )
+    private val DESCRIPTION_FQN = FullQualifiedName("diagnosis.Description")
+    private val LAT_FQN = FullQualifiedName("location.latitude")
+    private val LON_FQN = FullQualifiedName("location.longitude")
+    private val SHARING_NAME_FQN =FullQualifiedName("sharing.name")
+    private val VERSION_FQN = FullQualifiedName( "ol.version")
+    private val EMAIL_FQN = FullQualifiedName("contact.Email")
+
+    private val fqnToColumnName = mapOf(
+        STRING_ID_FQN to  "study_id",
+        FULL_NAME_FQN to "fullname",
+        DESCRIPTION_FQN to "description",
+        LAT_FQN to "latitude",
+        LON_FQN to "longitude",
+        SHARING_NAME_FQN to "study_group",
+        VERSION_FQN to "study_version",
+        EMAIL_FQN to "contact",
+        LAST_WRITE_FQN to "updated_at"
+    )
+
+    // private val GET_V2_STUDIES_SQL = """
+    //     SELECT id, ? AS ? FROM
+    //     (SELECT id, ? AS ? FROM data WHERE entity_set_id = ANY(?) AND property_type_id = ? ORDER BY id) AS ?
+    // """.trimIndent()
+
+    // private val V2_DATA_INNERJOIN_SQL = """
+    //     INNER JOIN
+    //     (SELECT id, ? AS ? FROM data WHERE entity_set_id = ANY(?) AND property_type_id = ? ORDER BY id) AS ?
+    //     USING (id)
+    // """.trimIndent()
+
+    // private val INSERT_INTO_STUDY_TABLE_SQL = """
+    //     INSERT INTO studies (id, fullname, description, latitude, longitude, sharing, version, email) VALUES (?,?,?,?,?,?,?,?)
+    // """.trimIndent()
+
+    override fun upgrade(): Boolean {
+        logger.info("starting migration of studies to v3")
+
+        // Property Types of ol.study
+        val studiesPropertyTypes = propertyTypes.entrySet(
+            Predicates.`in`<UUID, PropertyType>(
+                "id",
+                // "general.stringid",
+                UUID.fromString("ee3a7573-aa70-4afb-814d-3fad27cda988"),
+                // "general.fullname",
+                UUID.fromString("70d2ff1c-2450-4a47-a954-a7641b7399ae"),
+                // "diagnosis.Description",
+                UUID.fromString("c502df5b-ec63-4f15-b529-d38695366c75"),
+                // "location.latitude",
+                UUID.fromString("06083695-aebe-4a56-9b98-da6013e93a5e"),
+                // "location.longitude",
+                UUID.fromString("e8f9026a-2494-4749-84bb-1499cb7f215c"),
+                // "sharing.name",
+                UUID.fromString("54f64f11-9c4d-4701-897b-630ffeb94a21"),
+                // "ol.version",
+                UUID.fromString("0011bbfe-d5d4-4f88-97a8-cdeb821deb6f"),
+                // "contact.Email"
+                UUID.fromString("6a1f7cf6-80eb-4fe9-a9f4-49cad15c6154")
+            )
+        )
+            // need to do this silly conversion or entrySet will return Set<Map.Entry<,>>
+            .map { it.key to it.value }.toMap()
```

Comment:
I think you could just do this ...

```
val studiesPropertyTypes = propertyTypes
  .getAll(setOf(
    UUID.fromString(...),
    ...
  ))
  .toMap()
```

Reply:
much better, will do that instead, thanks!